src/realm/object-store/util/scheduler.cpp

```diff
@@ -28,7 +28,16 @@
 #include <realm/object-store/util/apple/scheduler.hpp>
 #endif
 
-#if REALM_ANDROID
+// When building Realm within the VNDK in AOSP, __ANDROID__ is defined.
+// However, access to libandroid is restricted by VNDK policies.
+// As a result, we cannot utilize the built-in ALooper functionality.
+// Instead, we require users to provide their own scheduler implementation.
+
+#if REALM_ANDROID && !defined(REALM_AOSP_VENDOR)
+#define HAS_ANDROID_ALOOPER
+#endif
+
+#if HAS_ANDROID_ALOOPER
```

Comment:
This expands either to
~~~
#if
~~~
or 
~~~
#if HAS_ANDROID_ALOOPER
~~~
causing 
~~~
/mnt/vcpkg-ci/b/realm-core/src/14.8.0-91fbc84ff4.clean/src/realm/object-store/util/scheduler.cpp:40:24: error: expected value in expression
#if HAS_ANDROID_ALOOPER
~~~

Reply:
Hi @dg0yt the latest master contains a fix for this where it becomes `#if defined(HAS_ANDROID_ALOOPER)`

Reply:
Thanks.